core/src/main/scala/io/chrisdavenport/fuuid/annotation/FUUID.scala

```diff
@@ -0,0 +1,107 @@
+package io.chrisdavenport.fuuid.annotation
+
+import scala.language.experimental.macros
+import scala.annotation.{compileTimeOnly, StaticAnnotation}
+import scala.reflect.macros.whitebox
+
+/**
+ * This annotation can be used on any kind of object to automatically
+ * create an inner `Id` tagged `FUUID` type with convenient methods for
+ * its creation. It also provides implicit instances for cats' `Hash`,
+ * `Order` and `Show` type-classes. All these instances are available
+ * in the enclosing object.
+ *
+ * @example For an object named `User` {{{
+ * object User {
+ *
+ *    trait IdTag
+ *
+ *    type Id = shapeless.tag.@@[FUUID, IdTag]
+ *
+ *    object Id {
+ *
+ *        //Creates a new `Id` from a `FUUID`
+ *        def apply(fuuid: FUUID): User.Id = ???
+ *
+ *        //Creates a new `Id` from an `UUID` literal. This method
+ *        //uses a macro to compile check the literal value
+ *        def apply(s: String): User.Id = ???
+ *
+ *
+ *        //Creates a random `Id` wrapped in an `F`
+ *        def random[F[_]: cats.effect.Sync]: F[User.Id] = ???
+ *
+ *        object Unsafe {
+ *
+ *          //Creates an unwrapped random `Id`
+ *          def random: User.Id = ???
+ *
+ *        }
+ *
+ *    }
+ *
+ *    implicit val IdHashOrderShowInstances: Hash[$name.Id] with Order[$name.Id] with Show[$name.Id] = ???
+ *
+ * }
+ * }}}
+ */
+@compileTimeOnly("enable macro paradise to expand macro annotations")
+class FUUID extends StaticAnnotation {
```

Comment:
Will this conflict on import?

Reply:
Mmmm, with [`FUUID`](https://github.com/ChristopherDavenport/fuuid/blob/880ccfffe028c6038293da2b67a06f0153a23041/modules/core/src/main/scala/io/chrisdavenport/fuuid/FUUID.scala#L11), yep.

Maybe we can think of another name for the annotation... `DeriveId`?

Reply:
Yeah, that seems reasonable.

Reply:
Updated to `@DeriveId` https://github.com/ChristopherDavenport/fuuid-annotation/pull/1/commits/ef5aabba83a7541273235515c9ee3dfeeb9f3d74